testify/test_case.py

```diff
@@ -168,25 +169,37 @@ def runnable_test_methods(self):
         any of our exclude_suites.  If there are any require_suites, it will then further
         limit itself to test methods in those suites.
         """
-        for member_name in dir(self):
-            if not member_name.startswith("test"):
-                continue
-            member = getattr(self, member_name)
-            if not inspect.ismethod(member):
-                continue
-
-            member_suites = self.suites(member)
-
-            # if there are any exclude suites, exclude methods under them
-            if self.__suites_exclude and self.__suites_exclude & member_suites:
-                continue
-            # if there are any require suites, only run methods in *all* of those suites
-            if self.__suites_require and not ((self.__suites_require & member_suites) == self.__suites_require):
-                continue
-
-            # if there are any name overrides, only run the named methods
-            if self.__name_overrides is None or member.__name__ in self.__name_overrides:
-                yield member
+        already_inspected = set()
+        name_overrides, self.__name_overrides = itertools.tee(self.__name_overrides)
```

Comment:
I'm -1 on this, I don't believe it makes it any more lazy than the left hand side and at a great cost to readability